File: packages/typescript/examples/01_hello_world.ts
/**
 * Simplest possible A2A Lite agent.
 *
 * Run with: npx ts-node examples/01_hello_world.ts
 */
import { Agent } from '../src';

const agent = new Agent({
  name: 'HelloWorld',
  description: 'A simple greeting agent',
});

agent.skill('greet', { description: 'Greet someone by name' }, async ({ name = 'World' }) => {
  return `Hello, ${name}!`;
});

agent.run({ port: 8787 });

File: packages/python/examples/01_hello_world.py
"""
Simplest possible A2A Lite agent.

Run with: python examples/01_hello_world.py
Test with: a2a-lite test http://localhost:8787 greet -p name=World
"""
from a2a_lite import Agent

agent = Agent(
    name="HelloWorld",
    description="A simple greeting agent",
)


@agent.skill("greet", description="Greet someone by name")
async def greet(name: str = "World") -> str:
    return f"Hello, {name}!"


if __name__ == "__main__":
    agent.run(port=8787)
